### PR TITLE
gdal: update to 3.4.0 + fix clang build

### PR DESCRIPTION
**Pull request by raedrizqie**
hmm.. i dont know why the compiling took too long

### CODE REVIEW

**Comment by raedrizqie** [comment]
mine compiled successfully here: https://github.com/raedrizqie/MINGW-packages/actions/runs/1468121900

File: mingw-w64-gdal/006-fix-compile-without-libtool.patch [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+ $(GDAL_LIB):	$(GDAL_OBJ) GDALmake.opt
+ 	rm -f libgdal.a
+ 	$(AR) r $(GDAL_LIB) $(GDAL_OBJ)
+ 	$(RANLIB) $(GDAL_LIB)
+ 
+-$(GDAL_SLIB):	$(GDAL_OBJ) $(GDAL_LIB)
+-	$(LD_SHARED) $(GDAL_SLIB_SONAME) $(GDAL_OBJ) $(GDAL_LIBS) $(LDFLAGS) $(LIBS) \
+-		-o $(GDAL_SLIB)
++$(GDAL_SLIB):	$(GDAL_LIB) $(GDAL_BIN_OBJ) $(GDAL_OBJ)
++	for f in $(GDAL_OBJ); do $(LD_SHARED) $$f $(GDAL_BIN_OBJ) $(GDAL_LIB) $(LDFLAGS) $(LIBS) -Wl,--out-implib,$(GDAL_SLIB_LINK) -o $(GDAL_SLIB) ; done
```
Comment (MehdiChinoune) [inline mingw-w64-gdal/006-fix-compile-without-libtool.patch:92]: I don't know why did you add this loop?
but this step took about 3 hours of buildtime (executed two times)

**Comment by MehdiChinoune** [comment]
Building gdal with clang is already fixed, We should wait for xerces to be fixed before enable it for gdal, without those tricks that introduce some anomalities.

File: mingw-w64-gdal/006-fix-compile-without-libtool.patch (hunk 1 above)
Comment (raedrizqie) [reply mingw-w64-gdal/006-fix-compile-without-libtool.patch:92]: i had to work around the error while compiling `command line is too long` (something like that) can you help me fix this?

**Comment by raedrizqie** [comment]
> Building gdal with clang is already fixed, We should wait for xerces to be fixed before enable it for gdal, without those tricks that introduce some anomalities.

ok then
